miden/tests/integration/operations/stack_ops.rs

```diff
@@ -14,17 +197,462 @@ fn swapdw() {
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn swapdw_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH)) {
+        let asm_op = "swapdw";
+        let mut expected_values = test_values[..WORD_LEN*2].to_vec();
+        let mut b = test_values[WORD_LEN*2..].to_vec();
+        expected_values.reverse();
+        b.reverse();
+        expected_values.append(&mut b);
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn movup_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH), movup_idx in 2_usize..MIN_STACK_DEPTH) {
+        let asm_op = format!("movup.{movup_idx}");
+        let mut expected_values = test_values.to_vec();
+        let movup_value = expected_values[MIN_STACK_DEPTH - movup_idx - 1];
+        expected_values.remove(MIN_STACK_DEPTH - movup_idx - 1);
+        expected_values.push(movup_value);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn movupw_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH), movupw_idx in 2_usize..WORD_LEN) {
+        let asm_op = format!("movupw.{movupw_idx}");
+        let idx1 = MIN_STACK_DEPTH - (movupw_idx + 1) * WORD_LEN;
+        let idx2 = MIN_STACK_DEPTH - movupw_idx * WORD_LEN;
+        let mut movupw_values = test_values[idx1..idx2].to_vec();
+        let mut expected_values = test_values[..idx1].to_vec();
+        expected_values.append(&mut test_values[idx2..].to_vec());
+        expected_values.append(&mut movupw_values);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn movdn_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH), movdn_idx in 2_usize..MIN_STACK_DEPTH) {
+        let asm_op = format!("movdn.{movdn_idx}");
+        let mut expected_values = test_values.to_vec();
+        let movdn_value = expected_values[MIN_STACK_DEPTH-1];
+        expected_values.remove(MIN_STACK_DEPTH-1);
+        expected_values.insert(MIN_STACK_DEPTH - movdn_idx - 1, movdn_value);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn movdnw_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH), movdnw_idx in 2_usize..WORD_LEN) {
+        let asm_op = format!("movdnw.{movdnw_idx}");
+        let idx1 = MIN_STACK_DEPTH - (movdnw_idx + 1) * WORD_LEN;
+        let mut movdnw_values = test_values[MIN_STACK_DEPTH-WORD_LEN..].to_vec();
+        let mut expected_values = test_values[..idx1].to_vec();
+        expected_values.append(&mut movdnw_values);
+        expected_values.append(&mut test_values[idx1..MIN_STACK_DEPTH-WORD_LEN].to_vec());
```

Comment:
For readability, I'd prefer if `MIN_STACK_DEPTH - WORD_LEN` were defined as a variable, especially since it's used twice.